### PR TITLE
Beacon hash sidechain sync test

### PR DESCRIPTION
**Pull request by aion-kelvin**
This is a test a specific code path in kernel: syncing a block that contains a transaction using beacon hash and that incoming block is attached to a side-chain.

Had to add a customized TestNodeManager to expose some functionality that didn't seem appropriate for the normal TestNodeManager.  Also, think there was a typo/pasting error in EventRequest, which I fixed here because my test needed it.

Take a look at the Javadoc and inline comments for info what the test does / why.

This does not run as part of concurrent suite and doesn't really work with the existing runners.  It needs to be run on its own, like the saturation tests.

### CODE REVIEW

File: Tests/test/org/aion/harness/tests/integ/runner/internal/LocalNodeListener.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
     /**
      * This method is invoked via reflection by the custom runner classes to initialize the listener.
      */
-    private void setListener(NodeListener listener) {
```
Comment (aionick) [inline Tests/test/org/aion/harness/tests/integ/runner/internal/LocalNodeListener.java]: We won't want to expose this. I think the issue here is that you are using `LocalNodeListener` for your tests but you shouldn't be. This class exists only as a java Rule that is used by our custom runners that manage the node's lifecycling internally (and hence, since tests that use these custom runners don't have access to a node, they can't do anything with a listener, this class addresses that issue).
In these kinds of tests, where you are responsible for the node's lifecycle yourself, you should be using `NodeListener` directly.

File: Tests/test/org/aion/harness/tests/integ/multikernel/BeaconHashSidechainTest.java [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+import org.aion.harness.main.RPC;
+import org.aion.harness.main.event.Event;
+import org.aion.harness.main.event.IEvent;
+import org.aion.harness.main.types.Block;
+import org.aion.harness.main.types.ReceiptHash;
+import org.aion.harness.main.types.TransactionReceipt;
+import org.aion.harness.result.FutureResult;
+import org.aion.harness.result.LogEventResult;
+import org.aion.harness.result.RpcResult;
+import org.aion.harness.tests.integ.runner.internal.LocalNodeListener;
```
Comment (aionick) [inline Tests/test/org/aion/harness/tests/integ/multikernel/BeaconHashSidechainTest.java]: See below comment, you should be using `NodeListener` instead

File: Tests/test/org/aion/harness/tests/integ/runner/internal/BeaconHashSidechainNodeManager.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
+package org.aion.harness.tests.integ.runner.internal;
```
Comment (aionick) [inline Tests/test/org/aion/harness/tests/integ/runner/internal/BeaconHashSidechainNodeManager.java]: This class probably shouldn't be in this package. This package is used for implementation details pertaining to our custom runners, but this class doesn't seem to be related to the runners at all.

**Review by aionick — CHANGES_REQUESTED** [review]

**Comment by aion-kelvin** [comment]
I've updated the commit with the requested changes

**Review by aionick — APPROVED** [review]
